File: Classes/crimson/ParseConfiguration.cpp
#include "ParseConfiguration.h"
#include "cocos2d.h"

#include "platform/android/jni/JniHelper.h"

#include <jni.h>

namespace dxco {

int ParseConfiguration::getAdMinLevel() {

	cocos2d::JniMethodInfo t;

	if (cocos2d::JniHelper::getStaticMethodInfo(t, "com/dxco/pandavszombies/ParseConfiguration", "getAdMinLevel", "()I")) {

		return (int) t.env->CallStaticIntMethod(t.classID, t.methodID);
	}

	return 15;
}

int ParseConfiguration::getShowAdRateSurvival() {

	cocos2d::JniMethodInfo t;

	if (cocos2d::JniHelper::getStaticMethodInfo(t, "com/dxco/pandavszombies/ParseConfiguration", "getShowAdRateSurvival", "()I")) {

		return (int) t.env->CallStaticIntMethod(t.classID, t.methodID);
	}

	return 15;
}

int ParseConfiguration::getShowAdRateStory() {

	cocos2d::JniMethodInfo t;

	if (cocos2d::JniHelper::getStaticMethodInfo(t, "com/dxco/pandavszombies/ParseConfiguration", "getShowAdRateStory", "()I")) {

		return (int) t.env->CallStaticIntMethod(t.classID, t.methodID);
	}

	return 15;
}

} /* namespace dxco */

File: proj.android/src/com/dxco/pandavszombies/ParseConfiguration.java
package com.dxco.pandavszombies;

import android.app.Activity;

public class ParseConfiguration {
	
	public static Activity activity;

	public static int getAdMinLevel() {
		return ConfigurationUtil.getConfiguration(activity, "showAdMinLevel", 15);
	}
	
	public static int getShowAdRateStory() {
		return ConfigurationUtil.getConfiguration(activity, "showAdRateStory", 34);
	}
	
	public static int getShowAdRateSurvival() {
		return ConfigurationUtil.getConfiguration(activity, "showAdRateSurvival", 34);
	}
	
	public static int getConfigurationTimeCacheHs() {
		return ConfigurationUtil.getConfiguration(activity, "configurationTimeCacheHs", 24);
	}
}
